src/qibocal/protocols/readout_mitigation_matrix.py

```diff
@@ -61,8 +62,9 @@ def _acquisition(
     data = ReadoutMitigationMatrixData(
         nshots=params.nshots, qubit_list=[list(qq) for qq in targets]
     )
-    backend = get_backend()
-    backend.platform = platform
+    os.environ["QIBO_PLATFORM"] = platform.name
+    backend = QibolabBackend(platform)
```

Comment:
Wait: if you directly use `QibolabBackend` [*], then `$QIBO_PLAFORM` should be irrelevant. Isn't it?

If that's the case, this should be sufficient
```suggestion
    backend = QibolabBackend(platform)
```

[*]: which is a great idea - until we'll reintroduce support in Qibocal for backends other than Qibolab #913 (but that's a separate issue, to tackle later on)

Reply:
> Wait: if you directly use `QibolabBackend` [*], then `$QIBO_PLAFORM` should be irrelevant. Isn't it?

That is also what I initially expected, however it seems that another instance of `GlobalBackend` is instantiated within the https://github.com/qiboteam/qibocal/blob/53a94cadfd254c9fec7be7e34250751a64fb6248/src/qibocal/protocols/readout_mitigation_matrix.py#L81
call which causes a failure if the environment variable is not properly set (which is the case if the slurm prologue is wrong). From the error trace, it seemed that this second `GlobalBackend` is related to the transpilation process, but I did not investigate in detail.

In any case, this PR was just a quick hotfix for something that @Luca-Ben-Herrmann was running, caused by the wrong slurm prologue of `qw5q_platinum` partition (which has also been fixed). Ideally, I would like to make all routines independent of the `GlobalBackend` to avoid such issues, but I am not sure how much of rework that requires.

> [*]: which is a great idea - until we'll reintroduce support in Qibocal for backends other than Qibolab #913 (but that's a separate issue, to tackle later on)

I guess hard-coding `QibolabBackend` here is also not optimal as it would not work for simulation, which I am not sure if is relevant for that particular routine. Anyway, this PR is mostly to raise these issues, maybe we don't even need to merge as it is, unless you are all fine with restricting with qibolab for now and introducing other backends later, as you are proposing.

Reply:
> I guess hard-coding `QibolabBackend` here is also not optimal as it would not work for simulation,

As said, I would not worry too much about this. Not yet.

> call which causes a failure if the environment variable is not properly set (which is the case if the slurm prologue is wrong). From the error trace, it seemed that this second `GlobalBackend` is related to the transpilation process, but I did not investigate in detail.

Actually, I failed to find it, since in that call the `backend` you're instantiating is passed in (and transitively passed down into the called `transpile_circuits()` function as well).